javascript/react/introduction-to-hooks/state-hooks.md

```diff
@@ -0,0 +1,136 @@
+---
+author: kapnobatai136
+
+levels:
+
+  - medium
+  
+aspects:
+
+  - new
+
+  - workout
+
+  - deep
+
+type: normal
+
+category: how to
+
+links:
+  - '[state-hooks](https://reactjs.org/docs/hooks-state.html){documentation}'
+
+---
+
+# State hooks
+
+---
+## Content
+
+Let's start off by looking at how you would usually initialize a state in a class component:
+
+```js
+class Enki extends React.Component(
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: "Enki"
+    };
+  }
+)
+```
+
+But what if you wanted to use functional components? In a function component you can't use the `this` keyword, nor can you assign or read `this.state`. This is where hooks come in:
+
+```js
+import React, { useState } from "react";
+
+function Enki() {
+  const [name, setName] = useState("Enki");
+}
+```
+
+Both of the examples above provide the same functionality. Note that you must import the `useState` hook from the `'react'` library.
+
+By calling the `useState()` hook we declare a *state variable*, which in our case is called `name`. This *state variable* can take any name you want, and it represents a way to preserve some values between the function calls. Basically, by calling `useState()` we enable the functionality of `this.state` for functional components.
+
+The only argument we pass to the `useState()` hook is the **initial state**, and if you want to store two different values in `state` you can call the `useState()` hook twice. The state doesn't have to an object, but it can be if you need it.
```

Comment:
For consistency, pick "we" or "you" (pick whatever you usually use or have seen in our content)

Reply:
agreed, I'll review the workout and make sure I am consistent